src/module/sheets/TwodsixActorSheet.ts

```diff
@@ -101,6 +111,85 @@ export class TwodsixActorSheet extends AbstractTwodsixActorSheet {
     await TwodsixRolls.performThrow(this.actor, itemId, dataset, showThrowDialog, numAttacks);
   }
 
+  private _onRollWrapper(func: (event:Event, showTrowDiag: boolean) => Promise<void>): (event:Event) => void {
+    return (event:Event) => {
+      event.preventDefault();
+      event.stopPropagation();
+
+      const useInvertedShiftClick:boolean = game.settings.get('twodsix', 'invertSkillRollShiftClick');
+      const showTrowDiag = useInvertedShiftClick ? event["shiftKey"] : !event["shiftKey"];
+
+      func.bind(this)(event, showTrowDiag);
+    };
+  }
+
+  /**
+   * Handle clickable weapon attacks.
+   * @param {Event} event   The originating click event
+   * @param {boolean} showTrowDiag  Whether to show the throw dialog or not
+   * @private
+   */
+  private async _onPerformAttack(event:Event, showTrowDiag: boolean):Promise<void> {
+    const attackType = event.currentTarget["dataset"].attackType;
+    const rof = event.currentTarget["dataset"].rof ? parseInt(event.currentTarget["dataset"].rof) : null;
+
+    const item = this.getItem(event);
+    await item.performAttack(attackType, showTrowDiag, rof);
+  }
+
+  /**
+   * Handle clickable skill rolls.
+   * @param {Event} event   The originating click event
+   * @param {boolean} showTrowDiag  Whether to show the throw dialog or not
+   * @private
+   */
+  private async _onSkillRoll(event:Event, showTrowDiag: boolean):Promise<void> {
+    const item = this.getItem(event);
+    await item.skillRoll(showTrowDiag);
+    // await TwodsixRolls.performThrow(this.actor, itemId, dataset, showThrowDialog, numAttacks);
+  }
+
+  /**
+   * Handle clickable characteristics rolls.
+   * @param {Event} event   The originating click event
+   * @param {boolean} showTrowDiag  Whether to show the throw dialog or not
+   * @private
+   */
+  private async _onRollChar(event:Event, showTrowDiag: boolean):Promise<void> {
+    await this.actor.characteristicRoll({"characteristic": $(event.currentTarget).data("label")}, showTrowDiag);
+  }
+
+  /**
+   * Handle clickable untrained skill rolls.
+   * @param {Event} event   The originating click event
+   * @param {boolean} showTrowDiag  Whether to show the throw dialog or not
+   * @private
+   */
+  private async _onRollUntrained(event:Event, showTrowDiag: boolean):Promise<void> {
+    const data = {
+      "name": game.i18n.localize("TWODSIX.Actor.Skills.Untrained"),
+      "data": {
+        "value": game.settings.get('twodsix', 'untrainedSkillValue')
+      },
+      "type": "skills"
+    };
+
+    // this feels very hacky, but creating temporary TwodsixItems only returns a plain js object..
+    // however the rest of the handling of this roll then behaves exactly like a normal skill, so it feels
+    // somewhat worth it.
+    let skill: TwodsixItem;
+    try {
+      skill = TwodsixItem.createOwned(data, this.actor) as TwodsixItem;
```

Comment:
Yeah, as the comment above says, this feels quite hacky. There are other ways to solve this, but the rest of the code becomes very nice by creating a temporary skill like this.